libs/single-spa-angular/elements/src/types.ts

```diff
@@ -2,8 +2,40 @@ import { NgModuleRef } from '@angular/core';
 import { NgElement } from '@angular/elements';
 import { BaseSingleSpaAngularOptions } from 'single-spa-angular/internals';
 
-export interface BootstrappedSingleSpaAngularElementsOptions extends BaseSingleSpaAngularOptions {
+// implementation from https://spin.atomicobject.com/2018/09/10/javascript-concurrency/
+export class Mutex {
+  private mutex = Promise.resolve();
+
+  lock(): PromiseLike<() => void> {
+    let begin: (unlock: () => void) => void;
+
+    this.mutex = this.mutex.then(() => {
+      return new Promise(begin);
+    });
+
+    return new Promise(resolve => {
+      begin = resolve;
+    });
+  }
+
+  async dispatch<T>(fn: (() => T) | (() => PromiseLike<T>)): Promise<T> {
```

Comment:
This types.ts file should only contain types, not implementations, but you've added implementations to it.